src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -538,21 +539,26 @@ public void acceptRecord(final SAMRecord record) {
                         boolean isOnTarget = false;
                         for (final Interval target : targets) {
                             if (refPos >= target.getStart() && refPos <= target.getEnd()) {
-                                isOnTarget = true;
                                 ++this.metrics.ON_TARGET_BASES;
                                 if (mappedInPair) ++this.metrics.ON_TARGET_FROM_PAIR_BASES;
 
                                 final int targetOffset = refPos - target.getStart();
                                 final Coverage coverage = this.coverageByTarget.get(target);
                                 coverage.addBase(targetOffset);
                                 baseQHistogramArray[baseQualities[offset]]++;
+                                coveredTargets.add(target);
+                                isOnTarget = true;
                             }
                         }
 
                         if (!isOnTarget) this.metrics.PCT_EXC_OFF_TARGET++;
                     }
                 }
             }
+            for (final Interval target : coveredTargets) {
```

Comment:
this for loop can be replaced with 
`if (!coveredTargets.contains(target)) coverage.incrementReadCount();`
or
```
if (!coveredTargets.contains(target)) {
  coverage.incrementReadCount();
  coveredTargets.add(target);
}
```

If the reason we are keeping distinct targets in `coveredTargets` is to keep a read count then it makes sense for the variables to live together.